### PR TITLE
Fix git error handling

### PR DESCRIPTION
**Pull request by wxiaoguang**

### CODE REVIEW

File: services/mirror/mirror_pull.go [hunk 1]
```diff
@@ -613,14 +613,8 @@ func checkAndUpdateEmptyRepository(ctx context.Context, m *repo_model.Mirror, re
...
-			if !git.IsErrUnsupportedVersion(err) {
-				log.Error("Failed to update default branch of underlying git repository %-v. Error: %v", m.Repo, err)
-				desc := fmt.Sprintf("Failed to update default branch of underlying git repository '%s': %v", m.Repo.RepoPath(), err)
-				if err = system_model.CreateRepositoryNotice(desc); err != nil {
-					log.Error("CreateRepositoryNotice: %v", err)
-				}
-				return false
-			}
+			log.Error("Failed to update default branch of underlying git repository %-v. Error: %v", m.Repo, err)
+			return false
```
Comment (lunny) [inline services/mirror/mirror_pull.go:617]: Maybe we should keep the notice here because the mirror is run in the backend.

Comment (wxiaoguang) [reply services/mirror/mirror_pull.go:617]: Who cares about the system notice? Who really ever used it?

Comment (lunny) [reply services/mirror/mirror_pull.go:617]: Actually, I will and some reports come from the system notice.

Comment (wxiaoguang) [reply services/mirror/mirror_pull.go:617]: Have you ever seen "Failed to update default branch of underlying git repository '%s': %v" ? And what would you do if you see it?

Comment (lunny) [reply services/mirror/mirror_pull.go:617]: Never. :)

**Review by lunny — APPROVED** [review]

**Review by bohde — APPROVED** [review]

**Pull request merged**